bin/reth/src/node/mod.rs

```diff
@@ -152,8 +156,15 @@ impl Command {
         // Does not do anything on windows.
         raise_fd_limit();
 
+        // Create a new Arc with the modified ChainSpec
+        let chain_spec_with_gas_limit = Arc::new(
+            Arc::try_unwrap(self.chain.clone())
+                .expect("Expected a single reference to ChainSpec")
+                .chainspec_with_block_gas_limit(self.block_gas_limit),
```

Comment:
This is incorrect and it makes the node unable to sync. Refer to CI.

Per above comment, this can just be removed.